src/Symfony/Component/String/Tests/LazyStringTest.php

```diff
@@ -29,47 +29,45 @@ public function testLazyString()
         $this->assertSame(1, $count);
     }
 
-    public function testLazyCallable()
-    {
-        $count = 0;
-        $s = LazyString::fromCallable([function () use (&$count) {
-            return new class($count) {
-                private $count;
-
-                public function __construct(int &$count)
-                {
-                    $this->count = &$count;
-                }
-
-                public function __invoke()
-                {
-                    return ++$this->count;
-                }
-            };
-        }]);
-
-        $this->assertSame(0, $count);
-        $this->assertSame('1', (string) $s);
-        $this->assertSame(1, $count);
-        $this->assertSame('1', (string) $s); // ensure the value is memoized
-        $this->assertSame(1, $count);
-    }
-
     /**
      * @runInSeparateProcess
      */
     public function testReturnTypeError()
     {
         ErrorHandler::register();
 
-        $s = LazyString::fromCallable(function () { return []; });
+        $s = LazyString::fromCallable(fn() => []);
 
         $this->expectException(\TypeError::class);
         $this->expectExceptionMessage('Return value of '.__NAMESPACE__.'\{closure}() passed to '.LazyString::class.'::fromCallable() must be of the type string, array returned.');
 
         (string) $s;
     }
 
+    public function testLazyCallable()
+    {
+        $count = 0;
+        $s = LazyString::fromCallable([fn() => new class($count) {
+            private $count;
+
+            public function __construct(int &$count)
+            {
+                $this->count = &$count;
+            }
+
+            public function __invoke()
+            {
+                return ++$this->count;
+            }
+        }]);
+
+        $this->assertSame(0, $count);
+        $this->assertSame('1', (string) $s);
+        $this->assertSame(1, $count);
```

Comment:
@nicolas-grekas Hello wave friendly ping as your help would be quite useful here.

When using an arrow function instead of a regular anonymous function, this assertion fails. I guess it's not expected as the signature allow all form of callables right ? If you confirm, I'll rollback these changes and open a new PR to try to fix this and extend tests to all forms of callables. A pointer as to how it could be fixed would be welcome in that case, thanks slightly_smiling_face

Otherwise I'll just rollback these changes and add some details about the exclusion of arrow functions callable in the parameter doc.

Reply:
I think it is because

> Values from the outer scope cannot be modified by arrow functions

(from https://www.php.net/manual/en/functions.arrow.php)

IMHO this test should be kept as-is.

Reply:
Yeah, I'll leave as it is then since migrating to arrow function in this case changes the currently expected behavior.

I'll let someone open a related a issue / PR if supporting arrow function with lazy string matters to them.